arviz/utils/xarray_utils.py

```diff
@@ -8,40 +8,70 @@
 from arviz.compat import pymc3 as pm
 
 
-def convert_to_xarray(obj, coords=None, dims=None):
-    """Convert a supported object to an xarray dataset.
+def get_converter(obj, coords=None, dims=None, chains=None):
+    """Get the converter to transform a supported object to an xarray dataset.
 
     This function sends `obj` to the right conversion function. It is idempotent,
     in that it will return xarray.Datasets unchanged.
 
     Parameters
     ----------
-    obj : An object from PyStan or PyMC3 to convert
+    obj : A dict, or an object from PyStan or PyMC3 to convert
     coords : dict[str, iterable]
         A dictionary containing the values that are used as index. The key
         is the name of the dimension, the values are the index values.
     dims : dict[str, Tuple(str)]
         A mapping from pymc3 variables to a tuple corresponding to
         the shape of the variable, where the elements of the tuples are
         the names of the coordinate dimensions.
+    chains : int or None
+        The number of chains sampled from the posterior, only necessary for
+        converting dicts.
 
     Returns
     -------
     xarray.Dataset
         The coordinates are those passed in and ('chain', 'draw')
     """
-
-    if isinstance(obj, xr.Dataset):
-        return obj
+    if isinstance(obj, dict):
```

Comment:
It looks like this would throw a `TypeError` if it was given an `xarray.Dataset` object.  I'm not sure if there's an elegant way to avoid that, but tests *should* fail if these functions are not idempotent.

Reply:
This should be fixed now.